### PR TITLE
Add support for callers with different future date range and item count limits

### PR DESCRIPTION
**Pull request by sdetweil**
####  Description
the default calendar service selects up to 9 entries from 9 future days. my plugin needs to allow variable number of days and items.  these changes allow callers to specify their unique event storage location, and days in advance and number of items to return

####  Fixes Related issues
- add related
- issues here

####  Checklist
- [x] I have read and understand the CONTRIBUTIONS.md file
- [x] I have searched for and linked related issues
- [x] I have added config.schema.json file if config option are required.
- [x] I am NOT targeting master branch

### CODE REVIEW

**Comment by justbill2020** [comment]
@sdetweil, 

not sure if this is useful to all users... honestly not sure what this is doing. 

Can you explain it a bit more about what's happening here, I would like to merge this if it serves the community but i'm just not sure. I don't mean to be overly critical or anything its just once its merged we need to be able to support and maintain the code. any added explainations would be great thank you.

**Comment by sdetweil** [comment]
sure.. the original code had a hard coded buffer to put the events in, 
```
	function CalendarService($window, $http, $q) {
		var service = {};

		service.events = [];  <-----
```

and the internal code used the settings in the config.json for the number of days and number of events to return .. ```		service.getFutureEvents = function () {
			var future_events = [],
				current_date = new moment(),
				end_date = new moment().add(config.calendar.maxDays, 'days');  <---------

			service.events.forEach(function (itm) {
        //If the event started before current time but ends after the current time or
        // if there is no end time and the event starts between today and the max number of days add it.
				if ((itm.end != undefined && (itm.end.isAfter(current_date) && itm.start.isBefore(current_date))) || itm.start.isBetween(current_date, end_date)) {
					future_events.push(itm);
				}
			});
			future_events = sortAscending(future_events);
			return future_events.slice(0, config.calendar.maxResults);  <------
		}```
which means I cannot ask for different parameters..   

also, if you have two processes using the same buffer, things will be bad eventually.  so, all I did was rework the routines to take that data as parameters from the routines caller..  nothing else changed.. 

I need access to the calendar entries in my plugin.. no point writing the whole thing again..

**Comment by sdetweil** [comment]
how did all those commits get into my pull request branch? looks like the stuff from the new master

**Comment by justbill2020** [comment]
we pulled dev to master last night and it is acting like dev is still 46 commits behind

**Comment by sdetweil** [comment]
ok, I reviewed the updated calendar branch, and there are no additional changes (from me) to this pull request..

**Comment by justbill2020** [comment]
correct

**Comment by evancohen** [comment]
If you could rebase `dev` onto your branch (or just merge it) then those extra commits will go away. Can you do that before I review?

**Comment by sdetweil** [comment]
ok, think I have done that

File: remote/index.html [hunk 1]
```diff
@@ -48,6 +48,21 @@
...
+                        </span>
+                        <span id="speak" class="input-group-addon">
+                            <i class="fa fa-microphone"></i>
+                        </span>
+                    </div>
+                </br>
+                <button id="command-bttn" type="button" class="btn btn-primary btn-lg btn-block glyphicons">Send Command!</button>
+            </div>
+        </div>
+        <div class="col-md-6 col-md-offset-3">
```
Comment (justbill2020) [inline remote/index.html]: duplicate code to lines 36-50 and needs to be resolved...

Comment (sdetweil) [inline remote/index.html]: that is not my file, and not changed in my PR..

**Comment by sdetweil** [comment]
remember my process is running on a timer tick, NOT inside the main process, so I have no idea when things get executed.. I would not count on it.. also, I have NO idea how long it will take me to process the info in the buffer.. so I would like my own buffer..

**Comment by justbill2020** [comment]
Ok then it sounds this is specific to your build and custom plugin and not useful to all users. Having said that perhaps we shouldn't merge this PR

**Comment by sdetweil** [comment]
it doesn't hurt you.. and it helps whoever comes later..  like me.. keeps the system footprint down.

**Comment by justbill2020** [comment]
@kurtdb can you take a look at this one too?

@evancohen has been super busy.

**Comment by kurtdb** [comment]
@justbill2020 what do you want me to do exactly? Review the code?

File: plugins/fitbit/config.schema.json [hunk 2]
```diff
@@ -82,7 +82,11 @@
       "type":"fieldset",
       "title":"Fitbit Settings",
       "expandable":true,
+<<<<<<< HEAD
```
Comment (evancohen) [inline plugins/fitbit/config.schema.json]: Please fix the schema. You merged some conflicts here.

File: plugins/calendar/service.js [hunk 3]
```diff
@@ -4,138 +4,148 @@
...
+							.replace(/\\r\\n/g, '<br />')
+							.replace(/\\n/g, '<br />')
+							.replace(/\\,/g, ',');
 					}
 
-          //Add the value to our event object.
-					if (type !== 'SUMMARY' || (type == 'SUMMARY' && cur_event['SUMMARY'] == undefined)) {
+					//Add the value to our event object.
+					if ( type !== 'SUMMARY' || (type=='SUMMARY' && cur_event['SUMMARY'] == undefined)) {
+
```
Comment (evancohen) [inline plugins/calendar/service.js]: nit: extra line

**Pull request closed without merging**